### PR TITLE
Review for DM-9002: Fix & document S/G code

### PR DESCRIPTION
**Pull request by jdswinbank**
The Star v/s Galaxy code inherited from Jose Garmilla has been updated
to make it runnable with recent versions of the LSST stack.

### CODE REVIEW

File: fsButler/utils.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import re
 import numpy as np
 import matplotlib.pyplot as plt
+import pdb
```
Comment (jdswinbank) [inline fsButler/utils.py]: Unused import.

File: fsButler/utils.py [hunk 2]
```diff
@@ -9,14 +10,15 @@
 import lsst.analysis.utils as utils
 import lsst.afw.display.ds9 as ds9
 
-from lsst.pex.exceptions import LsstCppException
-from lsst.meas.extensions.multiShapelet import FitPsfModel
+# from lsst.pex.exceptions import LsstCppException
```
Comment (jdswinbank) [inline fsButler/utils.py]: (Here and throughout) please don't comment out code and leave it in the file. If it's not needed, remove it.

File: fsButler/utils.py [hunk 3]
```diff
@@ -9,14 +10,15 @@
...
+from lsst.pex.exceptions import Exception as LsstCppException
+# from lsst.meas.extensions.multiShapelet import FitPsfModel
 
 """
 Utility functions to process data elements delivered by fsButler
 """
 
-#TODO: Make these configurable options
+# TODO: Make these configurable options
```
Comment (jdswinbank) [inline fsButler/utils.py:19]: (Here and throughout) whitespace churn should be on a separate commit, if, indeed, it's really necessary at all.

File: fsButler/utils.py [hunk 4]
```diff
@@ -59,29 +61,38 @@
 _zeroMagErrField = afwTable.Field["F"]("flux.zeromag.err",
                                        "The flux error corresponding to zero magnitude.")
 _stellarField = afwTable.Field["Flag"]("stellar",
-                                       "If true, the object is known to be a star if false it's known not to be a star.")
+                                       "If true, the object is known to be a star if false it's known not to\
+                                       be a star.")
```
Comment (jdswinbank) [inline fsButler/utils.py]: (Here and throughout) embedding new-lines in strings like this will cause weird spacing issues. Observe the difference:

```python
In [1]: a = ("123456789"
   ...:      "234567809")
   ...: print(a)
   ...:
123456789234567809

In [2]: b = ("123456789\
   ...:       234567809")
   ...: print(b)
   ...:
123456789      234567809
```

File: fsButler/utils.py [hunk 5]
```diff
@@ -315,14 +333,18 @@ def strictMatch(cat1, cat2, matchRadius=1*afwGeom.arcseconds, includeMismatches=
                 bestMatches[id] = (m1, m2, d)
 
     if includeMismatches:
-        print "{0} objects from {1} in the first catalog had no match in the second catalog.".format(len(noMatch), len(cat1))
-        print "{0} objects from the first catalog with a match in the second catalog were not the closest match.".format(len(matched) - len(noMatch) - len(bestMatches))
+        print "{0} objects from {1} in the first catalog had no match in the second\
+        catalog.".format(len(noMatch), len(cat1))
```
Comment (jdswinbank) [inline fsButler/utils.py]: (Here and below) in Python 3 (or Python 2 with `from __future__ import print_function`, which is the LSST standard), this is a syntax error.

File: fsButler/utils.py [hunk 6]
```diff
@@ -353,11 +376,12 @@ def matchMultiBand(butler, dataType, filters=['HSC-G', 'HSC-R', 'HSC-I', 'HSC-Z'
...
     for i in range(1, len(filters)):
         matched = strictMatch(matched, cats[i], multiMeas=multiMeas)
-    
+
     return matched
 
+
 def matchCats(cat1, cat2, matchRadius=1*afwGeom.arcseconds, includeMismatches=False,
-              multiMeas=False, suffix='.2'):
+              multiMeas=False, suffix='2'):
```
Comment (jdswinbank) [inline fsButler/utils.py:382]: The change to `suffix` seems kinda gratuitous. Is it necessary due to formatting of table field names?

File: fsButler/utils.py [hunk 7]
```diff
@@ -418,7 +448,11 @@ def matchCats(cat1, cat2, matchRadius=1*afwGeom.arcseconds, includeMismatches=Fa
         if name not in schema1.getNames():
             catKeys.append(schema.addField(schema2.find(name).getField()))
         else:
-            catKeys.append(schema.addField(schema2.find(name).getField().copyRenamed(name+suffix)))
+            if name == 'id':
+                catKeys.append(schema.addField(schema2.find(name).getField().copyRenamed(name + suffix)))
+            else:
+                newname = name.split('_')[0] + suffix + '_' + name.split('_')[1]
```
Comment (jdswinbank) [inline fsButler/utils.py]: Better to use `schema.join()` rather than hard-code `_`s?

File: fsButler/utils.py [hunk 8]
```diff
@@ -722,13 +772,16 @@ def getParent(objId, fsButler, dataType='deepCoadd'):
...
-       multIds[b] = cat.get('multId.'+b)
-   return multIds
+    bands = getCatBands(cat)
+    multIds = np.zeros((len(cat),), dtype=[(b, 'int64') for b in bands])
+    for b in bands:
+        multIds[b] = cat.get('multId.'+b)
+    return multIds
 
+
+''' Commented out since we don't need to perform the explaratory analysis analysis anymore.
```
Comment (jdswinbank) [inline fsButler/utils.py]: Typo (“explaratory analysis analysis”).

Prefer removing code to commenting it out. Obviously, that should be on a separate commit, and the commit message should explain what the analysis in question was.

File: fsButler/utils.py (hunk 1 above)
Comment (AstroVPK) [reply fsButler/utils.py]: Removed.

File: fsButler/utils.py (hunk 2 above)
Comment (AstroVPK) [reply fsButler/utils.py]: Done.

File: fsButler/utils.py (hunk 3 above)
Comment (AstroVPK) [reply fsButler/utils.py:19]: Ok. Duly noted.

File: fsButler/utils.py (hunk 6 above)
Comment (AstroVPK) [reply fsButler/utils.py:382]: Yes, it is necessary.

File: fsButler/utils.py (hunk 5 above)
Comment (AstroVPK) [reply fsButler/utils.py]: Fixed.

File: fsButler/utils.py (hunk 8 above)
Comment (AstroVPK) [reply fsButler/utils.py]: Removed...
